test/test/util/fake_segment_prefetch.js

```diff
@@ -0,0 +1,81 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * A fake SegmentPrefetch class that is used for testing
+ * segment prefetching functionality.
+ *
+ * @final
+ * @struct
+ * @extends {shaka.media.SegmentPrefetch}
+ */
+shaka.test.FakeSegmentPrefetch = class {
+  constructor(stream, segmentData) {
+    /** @private {(Set.<!shaka.media.SegmentReference>)} */
+    this.requestedReferences_ = new Set();
+
+    /** @private {shaka.extern.Stream} */
+    this.streamObj_ = stream;
+
+    /** @private {
```

Comment:
The `/**` and `*/` should be on separate lines. Also, this could be compacted some.

```js
   /**
    * @private {!Object.<string, shaka.test.FakeMediaSourceEngine.SegmentData>}
    */
```